src/libraries/Microsoft.PowerFx.Core/Public/Values/FileValue.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Buffers.Text;
+using System.Globalization;
+using System.Text;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+
+namespace Microsoft.PowerFx.Types
+{
+    public abstract class FileValue : ValidFormulaValue
+    {
+        private readonly string _string = null;
+        private readonly bool _isBase64Encoded = false;
+        private readonly FileType _fileType;
+        private readonly string _identifier;
+        private readonly int _id;
+        private readonly ResourceManager _resourceManager;
+
+        public FileType FileType => _fileType;
+
+        public int Id => _id;
+
+        internal FileValue(ResourceManager resourceManager, string str, bool isBase64Encoded, FileType fileType)
```

Comment:
>string str, bool isBase64Encoded [](http://example.com/codeflow?start=60&length=32)

I really don't think that we (Power Fx) should be in the business of managing file *content* - this should be handled via some sort of resource manager that is handled by the host.

Reply:
Makes sense, we could move the content to the resource manager

Reply:
From discussion w/ Carlos: this constructor could just take an IResourceManager and Id
Real content would be managed in RM (managed by host)